src/plugins/beforeInputPlugin.ts

```diff
@@ -63,8 +63,10 @@ export function beforeInputPlugin(
           // current cursor. We'll restore this later, so that React
           // doesn't panic about unknown DOM nodes.
           const { node: parent } = view.domAtPos($pos.pos);
+          precompositionSnapshot = [];
           parent.childNodes.forEach((node) => {
-            precompositionSnapshot.push(node);
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            precompositionSnapshot!.push(node);
```

Comment:
`NodeList` has supported iteration for almost ten years, so you could use `for of` here and thereby avoid the closure and the corresponding non-null assertion because TypeScript should be able to follow the impossibility of it bein null here.